NitroxClient/Communication/Client.cs

```diff
@@ -43,26 +37,29 @@ public void Start(String ip)
 
                 if (!socket.Connected)
                 {
-                    outputMessage("Unable to connect to server.");
+                    ClientLogger.WriteLine("Unable to connect to server.");
                     throw new InvalidOperationException("Socket could not connect.");
                 }
                 else
                 {
-                    outputMessage("Connected to server.");
+                    ClientLogger.WriteLine("Connected to server.");
                 }
 
                 connection.BeginReceive(new AsyncCallback(DataReceived));
             }
             catch (Exception e)
             {
-                outputMessage("Unable to connect to server");
+                ClientLogger.WriteLine("Unforeseen error when connecting: " + e.GetBaseException());
+                throw new InvalidOperationException("Unknown error while connecting");
             }
         }
 
         public void Stop()
```

Comment:
General question about the changes here... Why move game-centric clean up logic here? This class seems to be concerned with basic socket communication here (i.e. receiving and sending packets). My concern with adding outside clean-up logic here is the obfuscation of the core goal of this class.